### DISCUSSION TITLE
My Filter is not print traceId in log

### DISCUSSION THREAD

**Discussion by zhangzx1996** [apache/skywalking#12551] [Q&A] [answered]
<img width="959" alt="image" src="https://github.com/user-attachments/assets/6830c4a9-6a2b-435d-a004-3c7b23cef5f8">

![image](https://github.com/user-attachments/assets/a776cbe6-325e-4da8-b79d-2ce44c2d0db5)

I don't know why QAQ

**Comment by wu-sheng** [comment, marked as answer]
Are you using webflux? I think this question is being asked repeatedly due to you didn't search issue and read docs.

For webflux filter, you need to follow this, https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/application-toolkit-webflux/

**Reply by zhangzx1996** [reply]
Yes, I missed this doc. Very 3Q for you apply.
I was edit the filter . now I can see the trace.
<img width="747" alt="image" src="https://github.com/user-attachments/assets/af41f244-869a-41c6-963c-5318a96ce278">


But I meet a problem when I see the trace on skywalking OAP.
<img width="1678" alt="image" src="https://github.com/user-attachments/assets/bc6dfe1c-2af5-4c38-a348-a4b41ac87597">

I search more segments when the traceId is 0fffbbd22d3c464d97129c2fb725cabf.99.17243095037540081  
Why there are so many trace named "WebFluxOperators/onNext"

**Reply by wu-sheng** [reply]
It is a segment list, not a trace list. The title of the list is very clear about that.

**Reply by zhangzx1996** [reply]
But the traces of the ten segments are look same. And only gateway service occur this.
Next service is OK, just one trace segment, like this: 
 the traceId is 0fffbbd22d3c464d97129c2fb725cabf.99.17243095037540081
<img width="1681" alt="image" src="https://github.com/user-attachments/assets/a046cc12-2a58-4222-84ed-539aa62239b4">

**Reply by wu-sheng** [reply]
Segments just provide the link to the same trace. I don't say they are different. They belong to one trace if you query based on trace ID.
